taf/updater/updater.py

```diff
@@ -541,9 +545,21 @@ def _update_named_repository(
                         )
                         if error:
                             raise error
+                        return error
                     except Exception as e:
                         errors.append(str(e))
 
+                    # JMC: utilize ThreadPoolExecutor to run the update process in multiple threads
+                    with ThreadPoolExecutor() as executor:
```

Comment:
This is inside of `update_child_repo` and will never be called. I recommend testing this like this:
- create a new directory somewhere, say `test`
- inside that directory, run ` taf repo clone https://github.com/openlawlibrary/open-law --path openlawlibrary/open-law`
- delete test and run clone again to test again

Move this outside of this function, and it will start cloning these dependencies. You'll notice the difference between the outputs